slice-core-api/src/main/java/com/cognifide/slice/api/injector/InjectorConfig.java

```diff
@@ -57,14 +57,20 @@
 		// we don't allow to change the module list after creating the configuration
 		modules = Collections.unmodifiableList(new ArrayList<Module>(runner.getModules()));
 		name = runner.getInjectorName();
-		applicationPath = StringUtils.defaultIfEmpty(runner.getApplicationPath(), DEFAULT_INJECTOR_PATH
-				+ name);
+		applicationPaths = getApplicationPaths(runner);
 		parentName = runner.getParentName();
 		basePackage = runner.getBasePackage();
 		bundleFilter = runner.getBundleNameFilter();
 		listener = runner;
 	}
 
+	private List<String> getApplicationPaths(InjectorRunner runner) {
+		if (runner.getApplicationPaths() == null){
```

Comment:
I've analysed this deeper, and I don't think this `if` is correct. If somebody uses `InjectorRunner` without passing path(s) (`public InjectorRunner(final BundleContext bundleContext, final String injectorName, final String bundleNameFilter, final String basePackage)`), then this `if` will assign the injector to a default path (/apps/injectorName). This is not what happened in Slice 4.3. Basically, if path(s) is not specified by user, then we should not register any path to an injector. Please remove this and make sure that there will be no NPE in `InjectorHierarchy`